cirq/sim/simulator.py

```diff
@@ -525,3 +530,48 @@ def dirac_notation(self, decimals=2):
             A pretty string consisting of a sum of computational basis kets
             and non-zero floats of the specified accuracy."""
         return wave_function.dirac_notation(self.state, decimals)
+
+    def density_matrix(self, indices: List[int] = None) -> np.ndarray:
+        """Returns the sparse density matrix of the wavefunction.
+
+        Calculate the density matrix for the system on the given qubit
+        indices, with the qubits not in indices that are present in state
+        traced out. If indices is None the full density matrix for state
+        is returned.
+        For example:
+            state = np.array([1/np.sqrt(2), 1/np.sqrt(2)], dtype=np.complex64)
+            indices = None
+            gives us \rho = \begin{bmatrix}
```

Comment:
Is this Latex automatically rendered? Looking at: https://github.com/quantumlib/Cirq/issues/1073, it seems like there are a couple of doc strings in the repo that have Latex in them. An alternative is to type out the matrix similar to https://github.com/quantumlib/Cirq/blob/master/cirq/ops/parity_gates.py#L34; make it an easier read. I should add a comment on that issue, but @Strilanc, what is the plan on having Latex in the doc strings for people who won't generate the documentation to have it rendered?

Reply:
I think our audience is technical enough on average that we can expect them to read latex notation more precisely than whatever arbitrary text grid we use. So I'm fine with keeping it.